### PR TITLE
Handle SVG loading exceptions and path patching issues in IJ 233 EAP 4

### PR DESCRIPTION
**Pull request by rock3r**
Same as #188 but up-to-date compared to main.

### CODE REVIEW

File: core/src/main/kotlin/org/jetbrains/jewel/painter/ResourcePainterProvider.kt [hunk 1]
```diff
@@ -155,6 +159,23 @@ class ResourcePainterProvider(
         }
     }
 
+    @Composable
+    private fun createVectorDrawablePainter(url: URL, density: Density): VectorPainter {
+        val vector = url.openStream().use {
```
Comment (devkanro) [inline core/src/main/kotlin/org/jetbrains/jewel/painter/ResourcePainterProvider.kt:164]: Should we fallback to Magenta for bitmap and xml too?

File: core/src/main/kotlin/org/jetbrains/jewel/painter/ResourcePainterProvider.kt [hunk 2]
```diff
@@ -70,56 +76,31 @@ class ResourcePainterProvider(
...
                 }
                 add(path)
             }
         }.reversed()
 
-        val url = pathStack.firstNotNullOfOrNull {
-            resolveResource(it)
-        } ?: error("Resource '$basePath(${hints.joinToString()})' not found")
+        val url = pathStack.firstNotNullOfOrNull { resolveResource(it) }
+            ?: error("Resource '$basePath(${hints.joinToString()})' not found")
```
Comment (devkanro) [inline core/src/main/kotlin/org/jetbrains/jewel/painter/ResourcePainterProvider.kt:95]: Should we fallback to Magenta for unresolvable resource too?

**Review by devkanro — COMMENTED** [review]
Generally good, but the rules about fallback may still need to be discussed

File: core/src/main/kotlin/org/jetbrains/jewel/painter/ResourcePainterProvider.kt (hunk 1 above)
Comment (rock3r) [reply core/src/main/kotlin/org/jetbrains/jewel/painter/ResourcePainterProvider.kt:164]: You've got a good point. Will do!

File: core/src/main/kotlin/org/jetbrains/jewel/painter/ResourcePainterProvider.kt (hunk 2 above)
Comment (rock3r) [reply core/src/main/kotlin/org/jetbrains/jewel/painter/ResourcePainterProvider.kt:95]: You've got a good point. Will do!

File: core/src/main/kotlin/org/jetbrains/jewel/painter/ResourcePainterProvider.kt [hunk 3]
```diff
@@ -70,56 +81,32 @@ class ResourcePainterProvider(
...
```
Comment (devkanro) [inline core/src/main/kotlin/org/jetbrains/jewel/painter/ResourcePainterProvider.kt:100]: It still throw error in Non-Debug mode

**Review by devkanro — APPROVED** [review]
LGTM

**Review by lamba92 — APPROVED** [review]

**Pull request merged**